pkg/controller/atlasproject/private_endpoint.go

```diff
@@ -0,0 +1,250 @@
+package atlasproject
+
+import (
+	"context"
+	"errors"
+	"regexp"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.uber.org/zap"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/project"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/provider"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/util/set"
+)
+
+func (r *AtlasProjectReconciler) ensurePrivateEndpoint(ctx *workflow.Context, projectID string, project *mdbv1.AtlasProject) workflow.Result {
+	peList := project.Spec.DeepCopy().PrivateEndpoints
+	result, atlasPEs := createOrDeletePEInAtlas(ctx, projectID, peList, ctx.Log)
+	if !result.IsOk() {
+		return result
+	}
+
+	peStatus := convertToStatus(atlasPEs)
+	ctx.Log.Debugw("Private Endpoints Status", "status", peStatus)
+
+	ctx.EnsureStatusOption(status.AtlasProjectPrivateEnpointOption(peStatus))
+	return workflow.OK()
+}
+
+func createOrDeletePEInAtlas(ctx *workflow.Context, projectID string, operatorPrivateEndpoints []project.PrivateEndpoint, log *zap.SugaredLogger) (result workflow.Result, atlasPrivateEndpoints []AtlasProjectPrivateEndpoint) {
+	provider := "AWS"
```

Comment:
Why is this hardcoded?

Reply:
This was temporary.
The initial reason is that fetch all endpoints from all providers is considered outdated and fetch each provider separately is the preferred way. While developing for one provider I used this string for convenience.